### ISSUE TITLE
IntegrityError using Many-to-Many field with Postgres

### ISSUE THREAD

**Issue by johnthedebs** [bconstantin/django_polymorphic#9]
I'm using django-polymorphic for an app that has a models.py like this:

```
class Profile(PolymorphicModel):
    # fields...

class UserProfile(Profile):
    friends = models.ManyToManyField(Profile, related_name="followers")
    # more fields

class ArtistProfile(Profile):
    # fields...
```

If I do the following:

```
user = UserProfile.objects.get(pk=1)
artist = ArtistProfile.objects.get(pk=2)
user.friends.add(artist)
```

I get:

```
IntegrityError: insert or update on table "profiles_userprofile_friends" violates foreign key constraint "profiles_userprofile_friends_userprofile_id_fkey"
DETAIL:  Key (userprofile_id)=(2) is not present in table "profiles_userprofile".
```

But only when I'm using Postgres; it doesn't happen with SQLite (and I haven't tested it with any other DB).

The reverse works though, even with Postgres:

```
artist.followers.add(user)
```

**Comment by johnthedebs** [comment]
Nevermind, this seems to be an issue with Django 1.1 that's been fixed in 1.2. Nothing to see here.

http://code.djangoproject.com/ticket/11956

**Comment by bconstantin** [comment]
There seem to be some more differences between Postgres and MySQL/SQLite, so I'll probably start testing regularly with Postgres in any case.
